### DISCUSSION TITLE
Add support for 24bit

### DISCUSSION THREAD

**Discussion by joetoddsonos** [sonos/pyFLAC#17] [Ideas]
Currently the package only supports `int16` data format.

An `int24` type is not supported by `numpy` which makes this non-trivial with the current design.

**Comment by hughsw** [comment]
Newbie to using flac, and thank you for pyflac.
TL;DR  Can we at least allow pyflac to consume 24-bit streams, upcasting them to `int32`?

I need to decode flac streams that are encoded from 24-bit WAV format.  I see in warnings from flac that this is a questionable format:
`WARNING: legacy WAVE file has format type 1 but bits-per-sample=24`
However, `flac` generates a stream with a header that says it's 24 bits. And `flac -d` recreates the original WAV data (with slightly different header),

When I try to decode the stream using `pyflac.StreamDecoder`, I get an exception:
```
Traceback (most recent call last):
  File "/capyvenv/lib/python3.11/site-packages/pyflac/decoder.py", line 406, in _write_callback
    raise ValueError('Only int16/int32 data type is supported')
ValueError: Only int16/int32 data type is supported
```
Reading nearby comments I'm informed that the lower-level libraries are always returning 32-bit ints.  And indeed if the header says the data is 16-bit, then the code truncates the 32-bit values and returns an `int16` array.

I tweaked the code to allow a header that says 24-bit by just returning the 32-bit array.  Worked in my POC test bed -- the samples are preserved in the 32-bit array.

So, my question is, would allowing the upcast from 24-bit to 32-bit be a reasonable way to support consuming 24-bit data?  Should I post a PR with a proposed implementation?

**Reply by joetoddsonos** [reply]
Hi @hughsw, this seems like a reasonable approach, and is certainly better than having no support at all. Please do submit a PR, and I will take a look, thanks.
